wallet-new/src/Cardano/Wallet/WalletLayer.hs

```diff
@@ -563,3 +572,19 @@ instance Buildable RedeemAdaError where
         bprint ("RedeemAdaWalletIdDecodingFailed " % build) txt
     build (RedeemAdaInvalidRedemptionCode txt) =
         bprint ("RedeemAdaInvalidRedemptionCode " % build) txt
+
+data NewTransactionError =
+      NewTransactionError Kernel.NewTransactionError
+    | NewTransactionWalletIdDecodingFailed Text
+
+instance Show NewTransactionError where
+    show = formatToString build
```

Comment:
What about we let the compiler derives stock instances like `Show` and `Read` and never write them EVER again ^^" .. If the code uses `show` somewhere to convert the structure to string, that's definitely wrong. 
We should be able to copy/paste the output of `show` in ghci and end up with a value for that type.

Reply:
100%, this is an anti-pattern. @denisshevchenko generally speaking, unless there is a strong reason, `Show` and `Read` should abide the law `myType  = read . show` i.e. if you call show followed by a `read`, it should give you back the initial type. This is not law-abiding 😉

Reply:
Let's change this one as we are here -- do not define `Show` like that, use `Buildable` if you need to.